src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ServiceColletionExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.ContentManagement
+{
+    public static class ServiceColletionExtensions
+    {
+        /// <summary>
+        /// Registers a content part type.
+        /// </summary>
+        public static IServiceCollection AddContentPart<TContentPart>(this IServiceCollection serviceCollection)
+            where TContentPart : ContentPart
+        {
+            return serviceCollection.Configure<ContentOptions>(o => o.AddContentPart<TContentPart>());
+        }
+
+        /// <summary>
+        /// Registers a content field type.
+        /// </summary>
+        public static IServiceCollection AddContentField<TContentField>(this IServiceCollection serviceCollection)
+            where TContentField : ContentField
+        {
+            return serviceCollection.Configure<ContentOptions>(o => o.AddField<TContentField>());
+        }
```

Comment:
cool, so we can add to this with an Action if / when there becomes metadata to configure